### DISCUSSION TITLE
How to get current room name & (already) cleaned rooms?

### DISCUSSION THREAD

**Discussion by TheDelta** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#382] [Q&A] [answered]
I couldn't find any specific information, but I would like to use the attributes `cleaned_rooms` and `vacuum_room_name`.
They are always `null` for my Roborock S7 (`used_api: xiaomi`) and so far I can only successfully use `vacuum_room`.

My room numbers attribute looks like this:
```
room_numbers: 
'16': null
'17': null
'18': null
'19': null
'20': null
'21': null
'22': null
'23': null
```
(I assume instead of null the actual room name should be listed there?)

Is this not supported (yet) or am I missing something?

**Comment by PiotrMachowski** [comment, marked as answer]
Not every attribute is supported by every vacuum as maps for some models contain more data than other ones.

**Reply by TheDelta** [reply]
Thanks for the quick response. ❤️ 
Then I'll have to find some workaround for it to rely only on the `vacuum_room` number.
